libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -0,0 +1,213 @@
+			"Type conversion not yet implemented"
+		);
+		m_code <<
+			"let " <<
+			m_context.variableName(*_varDeclStatement.declarations().front()) <<
+			" := " <<
+			m_context.variable(*expression) <<
+			"\n";
+	}
+	else
+		for (auto const& decl: _varDeclStatement.declarations())
+			if (decl)
+				m_code << "let " << m_context.variableName(*decl) << "\n";
+
+	return false;
+}
+
+bool IRGeneratorForStatements::visit(Assignment const& _assignment)
+{
+	solUnimplementedAssert(_assignment.assignmentOperator() == Token::Assign, "");
+
+	_assignment.rightHandSide().accept(*this);
+
+//	solUnimplementedAssert(
+//		*_assignment.rightHandSide().annotation().type == *_assignment.leftHandSide().annotation().type,
+//		"Type conversion not yet implemented"
+//	);
+	// TODO proper lvalue handling
+	auto const& identifier = dynamic_cast<Identifier const&>(_assignment.leftHandSide());
+	string varName = m_context.variableName(dynamic_cast<VariableDeclaration const&>(*identifier.annotation().referencedDeclaration));
+	m_code << varName << " := " << m_context.variable(_assignment.rightHandSide()) << "\n";
+	m_code << "let " << m_context.variable(_assignment) << " := " << varName << "\n";
+	return false;
+}
+
+void IRGeneratorForStatements::endVisit(BinaryOperation const& _binOp)
+{
+	solUnimplementedAssert(_binOp.getOperator() == Token::Add, "");
+	solUnimplementedAssert(*_binOp.leftExpression().annotation().type == *_binOp.rightExpression().annotation().type, "");
+	if (IntegerType const* type = dynamic_cast<IntegerType const*>(_binOp.annotation().commonType.get()))
+	{
+		solUnimplementedAssert(!type->isSigned(), "");
+		m_code <<
+			"let " <<
+			m_context.variable(_binOp) <<
+			" := " <<
+			m_utils.overflowCheckedUIntAddFunction(type->numBits()) <<
+			"(" <<
+			m_context.variable(_binOp.leftExpression()) <<
+			", " <<
+			m_context.variable(_binOp.rightExpression()) <<
+			")\n";
+	}
+	else
+		solUnimplementedAssert(false, "");
+}
+
+bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
+{
+	solUnimplementedAssert(_functionCall.annotation().kind == FunctionCallKind::FunctionCall, "");
+	FunctionTypePointer functionType = dynamic_pointer_cast<FunctionType const>(_functionCall.expression().annotation().type);
+
+	TypePointers parameterTypes = functionType->parameterTypes();
+	vector<ASTPointer<Expression const>> const& callArguments = _functionCall.arguments();
+	vector<ASTPointer<ASTString>> const& callArgumentNames = _functionCall.names();
+	if (!functionType->takesArbitraryParameters())
+		solAssert(callArguments.size() == parameterTypes.size(), "");
+
+	vector<ASTPointer<Expression const>> arguments;
+	if (callArgumentNames.empty())
+		// normal arguments
+		arguments = callArguments;
+	else
+		// named arguments
+		for (auto const& parameterName: functionType->parameterNames())
+		{
+			bool found = false;
+			for (size_t j = 0; j < callArgumentNames.size() && !found; j++)
```

Comment:
Maybe ``std::find`` or something like that?

Reply:
Yep, should use `std::find`.